src/base/static/components/story-sidebar/index.js

```diff
@@ -0,0 +1,190 @@
+    if (!isRouteFound) {
+      this.setState({
+        currentStoryHeader: !this.isInitialized
+          ? storyConfig[this.stories.keySeq().first()].header
+          : "",
+        currentStoryDescription: !this.stories.isInitialized
+          ? storyConfig[this.stories.keySeq().first()].description
+          : "",
+        currentStoryName: !this.isInitialized
+          ? this.stories.keySeq().first()
+          : this.state.currentStoryName,
+        currentStoryChapters: !this.isInitialized
+          ? this.stories.first()
+          : this.state.currentStoryChapters,
+        currentStoryRoute: "",
+        currentChapter: Map(),
+      });
+    }
+  }
+
+  getModelFromUrl(url) {
+    const splitUrl = url.split("/");
+
+    // If the url has a slash in it with text on either side, we assume we have
+    // a place model url and can look up the model directly.
+    if (splitUrl.length === 2) {
+      return (
+        this.props.places[
+          this.props.layers.find(layer => layer.slug === splitUrl[0]).id
+        ].get(splitUrl[1]) || {}
+      );
+    } else {
+      // Otherwise, we have a "landmark-style" url, and have to search all place
+      // collections.
+      let foundModel;
+      Object.keys(this.props.places).forEach(collectionName => {
+        let model = this.props.places[collectionName].findWhere({
+          [constants.CUSTOM_URL_PROPERTY_NAME]: splitUrl[0],
+        });
+        if (model) {
+          foundModel = model;
+        }
+      });
+
+      return foundModel || {};
+    }
+  }
+
+  getIconUrl(chapter, storyUrl) {
+    // If the story config for this place declares an explicit sidebar icon url,
+    // use that icon.
+    if (
+      storyConfig[this.state.currentStoryName].order[storyUrl].sidebarIconUrl
+    ) {
+      return storyConfig[this.state.currentStoryName].order[storyUrl]
+        .sidebarIconUrl;
+    }
+
+    // If the story chapter has an icon url defined, use that.
+    if (chapter.get("style") && chapter.get("style").get("iconUrl")) {
+      return chapter.get("style").get("iconUrl");
+    }
+
+    // Otherwise, use the icon url defined in the category config.
+    return placeConfig.place_detail.find(
+      config => config.category === chapter.get("location_type"),
+    ).icon_url;
+  }
+
+  getTitle(chapter) {
+    // This is an unfortunate series of checks, but needed at the moment.
+    // TODO: We should revisit why this is necessary in the first place and see
+    // if we can refactor.
+    return (
+      chapter.get(constants.FULL_TITLE_PROPERTY_NAME) ||
+      chapter.get(constants.TITLE_PROPERTY_NAME) ||
+      chapter.get(constants.NAME_PROPERTY_NAME) ||
+      ""
+    );
+  }
+
+  render() {
+    return (
+      <div className="story-sidebar">
+        <a href="#" className="story-sidebar__collapse-btn" />
+        {this.state.currentStoryHeader && (
+          <h5 className="story-sidebar__header">
+            {this.state.currentStoryHeader}
+          </h5>
```

Comment:
nit: consider making the `a` tags and headers into atoms.

Reply:
Ah, I didn't realize atoms would include raw HTML tags as well. I think I get the idea behind atomic design, but what do we get from turning header and anchor tags into atoms? Aren't these already essentially atomic structures within the app?

Reply:
atoms are just small components. It's a pattern to easily manage consistency throughout our app. For example, if we have all `a` tags rendered through the same atom, let's say we call it Link, then we can ensure that the styling is handled all in one place. And we can always pass props to the Link component to override any behaviour.

It prevents duplication of CSS and styling.

Reply:
What was the decision on using the atoms here, instead of the `h5` and `a` tags directly? I see that we have these atoms implemented.

Although I can understand if we want to punt porting them over in another PR, or incrementally porting over our atoms.